src/main/java/com/celements/course/service/ICourseServiceRole.java

```diff
@@ -29,4 +29,7 @@ public DocumentReference getCourseTypeForCourse(DocumentReference courseDocRef)
 
   public boolean validateParticipant(DocumentReference regDocRef, String emailAdr,
       String activationCode);
+
+  public CourseConfirmState getConfirmeState(DocumentReference objDocRef);
```

Comment:
-  please choose better naming e.g. orderDocRef, registrationDocRef or regDocRef
-  add @Null or @NotNull to all parameter and return values in public methods at least in the interfaces/ComponentRoles